pkg/apis/autoscaling/v1/conversion.go

```diff
@@ -448,7 +449,14 @@ func Convert_v1_HorizontalPodAutoscaler_To_autoscaling_HorizontalPodAutoscaler(i
 
 	// drop round-tripping annotations after converting to internal
 	out.Annotations, _ = autoscaling.DropRoundTripHorizontalPodAutoscalerAnnotations(out.Annotations)
-
+	if out.Spec.Behavior != nil {
+		if out.Spec.Behavior.ScaleUp == nil {
```

Comment:
v2 defaults all fields inside scaleup/scaledown

This is only defaulting fields inside ScaleUp/ScaleDown if the top-level objects are nil.

Doesn't that means this code doesn't correctly default `autoscaling.alpha.kubernetes.io/behavior: {"ScaleUp":{}}`?

Reply:
Current test cases are not covering this. 
- Adding test cases will make sense.